src/utils.test.ts

```diff
@@ -52,18 +60,30 @@ describe("trimUriPrefix", () => {
 describe("parseSkylink", () => {
   it("should correctly parse skylink out of different strings", () => {
     const validSkylinkVariations = [
-      skylink,
-      `sia:${skylink}`,
-      `sia://${skylink}`,
-      `${portalUrl}/${skylink}`,
-      `${portalUrl}/${skylink}/`,
-      `${portalUrl}/${skylink}?`,
-      `${portalUrl}/${skylink}/foo/bar`,
-      `${portalUrl}/${skylink}?foo=bar`,
+      [skylink, ""],
+      [`${skylink}/`, "/"],
+      [`${skylink}//`, "//"],
+      [`${skylink}?`, "?"],
+      [`${skylink}?foo=bar`, "?foo=bar"],
+      [`${skylink}/foo/bar`, "/foo/bar"],
+      [`${skylink}#foobar`, "#foobar"],
+      [`sia:${skylink}`, ""],
+      [`sia://${skylink}`, ""],
+      [`${portalUrl}/${skylink}`, ""],
+      [`${portalUrl}/${skylink}/`, "/"],
+      [`${portalUrl}/${skylink}//`, "//"],
+      [`${portalUrl}/${skylink}?`, "?"],
+      [`${portalUrl}/${skylink}/foo/bar`, "/foo/bar"],
+      [`${portalUrl}/${skylink}/foo%3Fbar`, "/foo%3Fbar"],
+      [`${portalUrl}/${skylink}/foo/bar?foo=bar`, "/foo/bar?foo=bar"],
+      [`${portalUrl}/${skylink}?foo=bar`, "?foo=bar"],
+      [`${portalUrl}/${skylink}/?foo=bar`, "/?foo=bar"],
+      [`${portalUrl}/${skylink}#foobar`, "#foobar"],
+      [`${portalUrl}/${skylink}/#foobar`, "/#foobar"],
     ];
 
-    validSkylinkVariations.forEach((input) => {
-      expect(parseSkylink(input)).toEqual(skylink);
+    validSkylinkVariations.forEach(([fullSkylink, path]) => {
+      expect(parseSkylink(fullSkylink)).toEqual(`${skylink}${path}`);
```

Comment:
`parseSkylink` should return just the raw skylink, without the path, otherwise it's a breaking change and we will need another function that does that same thing anyway since I have my use cases for that

I would suggest adding a function argument like `includePath` (that's false by default) that would return the skylink with the path.

Also I don't think we should be adding query parameters or hash values. Those are not part of the skylinks.

Reply:
Okay I've addressed your points